### PR TITLE
Add message to api exception and update JavaDocs for error codes

### PR DESCRIPTION
**Pull request by alexanderjordanbaker**

### CODE REVIEW

File: src/main/java/com/apple/itunes/storekit/client/APIException.java [hunk 1]
```diff
@@ -9,40 +9,59 @@
...
     public int getHttpStatusCode() {
         return httpStatusCode;
     }
 
     public APIError getApiError() {
-        return apiError != null ? APIError.fetchErrorResponseFromErrorCode(apiError) : null;
+        return apiErrorCode != null ? APIError.fetchErrorResponseFromErrorCode(apiErrorCode) : null;
     }
 
     public Long getRawApiError() {
```
Comment (izanger) [inline src/main/java/com/apple/itunes/storekit/client/APIException.java:51]: Suggest renaming to `getRawApiErrorCode` or just `getApiErrorCode`

File: src/main/java/com/apple/itunes/storekit/client/APIException.java [hunk 2]
```diff
@@ -9,40 +9,59 @@
...
+        return apiErrorMessage;
     }
 
     @Override
     public String toString() {
         return "APIException{" +
                 "httpStatusCode=" + httpStatusCode +
-                ", apiError=" + apiError +
-                "} " + super.toString();
+                ", apiError=" + apiErrorCode +
```
Comment (izanger) [inline src/main/java/com/apple/itunes/storekit/client/APIException.java:63]: apiError -> apiErrorCode

File: src/main/java/com/apple/itunes/storekit/client/APIError.java [hunk 3]
```diff
@@ -10,60 +10,390 @@
...
+     * @see <a href="https://developer.apple.com/documentation/appstoreserverapi/invalidextendreasoncodeerror">InvalidExtendReasonCodeError</a>
+     */
     INVALID_EXTEND_REASON_CODE(4000010L),
+
+    /**
+     * An error that indicates an invalid request identifier.
+     *
+     * @see <a href="https://developer.apple.com/documentation/appstoreserverapi/invalidrequestidentifiererror">InvalidRequestIdentifierError</a>
+     */
     INVALID_IDENTIFIER(4000011L),
```
Comment (izanger) [inline src/main/java/com/apple/itunes/storekit/client/APIError.java:67]: Suggest a rename to `INVALID_REQUEST_IDENTIFIER`

File: src/main/java/com/apple/itunes/storekit/client/APIException.java (hunk 1 above)
Comment (alexanderjordanbaker) [reply src/main/java/com/apple/itunes/storekit/client/APIException.java:51]: This is getting the raw apiError enum, hence the name

File: src/main/java/com/apple/itunes/storekit/client/APIException.java (hunk 2 above)
Comment (alexanderjordanbaker) [reply src/main/java/com/apple/itunes/storekit/client/APIException.java:63]: The public name for this field is apiError

**Review by izanger — APPROVED** [review]

**Pull request merged**